simulator/dummy_perception_publisher/src/signed_distance_function.cpp

```diff
@@ -47,18 +50,23 @@ double AbstractSignedDistanceFunction::getSphereTracingDist(
 
 double BoxSDF::operator()(double x, double y) const
 {
-  const auto vec_global = tf2::Vector3(x, y, 0.0);
-  const auto vec_local = tf_global_to_local_.inverse()(vec_global);
+  const auto && vec_global = tf2::Vector3(x, y, 0.0);
+  const auto vec_local = tf_local_to_global_(vec_global);
 
   // As for signed distance field for a box, please refere:
   // https://www.iquilezles.org/www/articles/distfunctions/distfunctions.htm
   const auto sd_val_x = std::abs(vec_local.getX()) - 0.5 * length_;
   const auto sd_val_y = std::abs(vec_local.getY()) - 0.5 * width_;
   const auto positive_dist_x = std::max(sd_val_x, 0.0);
   const auto positive_dist_y = std::max(sd_val_y, 0.0);
-  const auto positive_dist = std::hypot(positive_dist_x, positive_dist_y);
+
+  const auto positive_dist =
```

Comment:
@HiroIshida 
[Q]
this line is equal to `const auto positive_dist = std::hypot(positive_dist_x, positive_dist_y);` so you dont have to change right?

Reply:
@taikitanaka3 
>> if you round the digits
Do you mean, using float is better?  

By the way, I fixed this one to original hypot. (I thought hypot is significantly slower than the hard-coding because std::hypot does some NAN and inf check, but when I compare using a benchmark, the performance is almost the same https://github.com/HiroIshida/snippets/blob/master/cpp/std_example/hypot_perf.cpp

Reply:
I compared code below and computation time doesn't different much 
```TEST(test, hypot) {
  const long double a = 1e32;
  const long double b = 1e32;
  // 0ms
  for (long long int i = 0; i < 1e32/*number of test*/; i++)
  {
    const long double c = std::hypot(a,b);
  }
}

TEST(test, sqrt_pow2) {
  const long double a = 1e32;
  const long double b = 1e32;
  // 0ms
  for (long long int i = 0; i < 1e32/*number of test*/; i++)
  {
    const long double c = std::sqrt(a*a+b*b);
  }
}
```

but in your bench mark code you are incrementing calculated value every time and that can also increase the time to compute

Reply:
anyway this is just reference